source/common/stats/stats_matcher_impl.cc

```diff
@@ -0,0 +1,53 @@
+#include "common/stats/stats_matcher_impl.h"
+
+#include <regex>
+#include <string>
+
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Stats {
+
+StatsMatcherImpl::StatsMatcherImpl(const envoy::config::metrics::v2::StatsConfig& config) {
+  switch (config.stats_matcher().stats_matcher_case()) {
```

Comment:
This is the kind of UI code that I think would be ideally shared between multiple applications of StringMatcher.

Reply:
Agreed -- I will add a TODO() to myself to factor this out, since StatsMatcherImpl is really just a very, very thin wrapper around what could be called a StringMatcherList?